service/prometheus/scrapeconfig.go

```diff
@@ -53,116 +56,223 @@ func getTarget(service v1.Service) model.LabelSet {
+					TargetLabel: AppLabel,
+					Replacement: CadvisorAppName,
 				},
-				// Drop any targets that don't match the service port regexp.
+				// Add cluster_id label.
+				clusterIDLabelRelabelConfig,
+				// Add cluster_type label.
+				clusterTypeLabelRelabelConfig,
+				// Add ip label.
+				ipLabelRelabelConfig,
+				// Add role label.
+				roleLabelRelabelConfig,
+				missingRoleLabelRelabelConfig,
+			},
+		},
+
+		{
+			JobName:                getJobName(service, KubeletJobType),
+			Scheme:                 HttpsScheme,
+			HTTPClientConfig:       insecureHTTPClientConfig,
+			ServiceDiscoveryConfig: nodeSDConfig,
+			RelabelConfigs: []*config.RelabelConfig{
+				// Add app label.
 				{
-					SourceLabels: model.LabelNames{PrometheusServicePortLabel},
-					Regex:        EndpointPortRegexp,
-					Action:       config.RelabelKeep,
+					TargetLabel: AppLabel,
+					Replacement: KubeletAppName,
 				},
+				// Add cluster_id label.
+				clusterIDLabelRelabelConfig,
+				// Add cluster_type label.
+				clusterTypeLabelRelabelConfig,
+				// Add ip label.
+				ipLabelRelabelConfig,
+				// Add role label.
+				roleLabelRelabelConfig,
+				missingRoleLabelRelabelConfig,
 			},
 		},
+
 		{
-			JobName:          getJobName(service, "cadvisor"),
-			Scheme:           HttpsScheme,
-			HTTPClientConfig: httpClientConfig,
-			ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
-				KubernetesSDConfigs: []*config.KubernetesSDConfig{
-					{
-						APIServer: apiServer,
-						Role:      config.KubernetesRoleNode,
-						TLSConfig: kubernetesTlsConfig,
+			JobName:                getJobName(service, NodeExporterJobType),
+			Scheme:                 HttpScheme,
+			ServiceDiscoveryConfig: endpointSDConfig,
+			RelabelConfigs: []*config.RelabelConfig{
+				// Only keep node-exporter endpoints.
+				{
+					SourceLabels: model.LabelNames{
+						KubernetesSDNamespaceLabel,
+						KubernetesSDServiceNameLabel,
 					},
+					Regex:  NodeExporterRegexp,
+					Action: config.RelabelKeep,
+				},
+				// Relabel address to node-exporter port.
+				{
+					SourceLabels: model.LabelNames{model.AddressLabel},
+					Regex:        KubeletPortRegexp,
+					Replacement:  NodeExporterPort,
+					TargetLabel:  model.AddressLabel,
+				},
+				// Add app label.
+				{
+					TargetLabel: AppLabel,
+					Replacement: NodeExporterAppName,
+				},
+				// Add cluster_id label.
+				clusterIDLabelRelabelConfig,
+				// Add cluster_type label.
+				clusterTypeLabelRelabelConfig,
+				// Add ip label.
+				{
+					SourceLabels: model.LabelNames{model.AddressLabel},
+					Regex:        NodeExporterPortRegexp,
+					Replacement:  GroupCapture,
+					TargetLabel:  IPLabel,
 				},
 			},
+		},
+
+		{
+			JobName:                getJobName(service, WorkloadJobType),
+			Scheme:                 HttpScheme,
+			ServiceDiscoveryConfig: endpointSDConfig,
 			RelabelConfigs: []*config.RelabelConfig{
-				clusterIDRelabelConfig,
+				// Only keep kube-state-metrics targets.
```

Comment:
One notable difference - on host clusters, we want to scrape all the general workload targets. On guest clusters, we only want to scrape our targets, so this is a whitelist here.